src/domain/Layouts/Inline/Inline.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { ReactElement } from 'react'
+
+import { Props, Variables } from '../../../common'
+import { StyledInlineElement, StyledInlineGroup, StyledInlineGroupWrapper } from './style'
+import { Item } from './Item'
+
+interface IInlineProps {
+  /** Component context */
+  componentContext?: string
+  spacing?: Variables.Spacing
+  align?: Align
+  collapse?: boolean
+  margins?: Props.IMargins
+}
+
+enum Align {
+  Left = 'left',
+  Right = 'right',
+  Center = 'center'
+}
+
+const Inline: React.FC<IInlineProps> & {Align: typeof Align, Item: typeof Item} = ({
+  componentContext,
+  spacing = Variables.Spacing.sXSmall,
+  align = Align.Left,
+  collapse= false,
+  margins,
+  children
+}) => {
+  const wrappedChildren = React.Children.map(
+    children,
+    (child: any) => {
+      let childProps = {}
+
+      if (child && child?.type && child?.type?.name === Item.name) {
+        childProps = child?.props
+      }
```

Comment:
This is definitely kind of weird but I think it provides the easiest experience when actually using the component (magically being able to wrap anything in `<Inline.Item fill>` and have it work. No clue how to get the typing to work (well I kind of do but I think there might be a better way) so I've just left it as `any` for now.

Reply:
may be add an interface for InlineItem?